operators/tokenizer/bpe_utils.hpp

```diff
@@ -101,9 +103,9 @@ class TokenWithRegularExp {
     m_text = val;
   }
 
-  std::pair<bool, std::u32string_view> GetNextToken() {
+  std::pair<bool, std::u32string_view> GetNextToken(std::string & regex_expr) {
```

Comment:
Can you write a unit test for this function, especially when there will be more patterns added?

Reply:
done! will add more after adding more functionality in a different PR so users can parse any regex, such as the one long one you sent me earlier.